server/bin/pbench-index.py

```diff
@@ -292,6 +309,11 @@ def main(options, name):
             skipped = Path(tmpdir, f"{name}.{idxctx.TS}.skipped")
             ie_filepath = Path(tmpdir, f"{name}.{idxctx.TS}.indexing-errors.json")
 
+            # Establish signal handler for INT, SIGHUP, TERM.
+            # signal.signal(signal.SIGHUP, handler)
+            signal.signal(signal.SIGINT, sigint_handler)
+            signal.signal(signal.SIGTERM, sigterm_handler)
```

Comment:
Once you establish these signal handlers, they are enabled for the rest of the time the process is running.  This means that we can get weird exceptions raised in contexts which we don't catch them.

Let's consider other implementation alternatives.

Reply:
there is already a try block in l.308. won't that work?

Reply:
See PR https://github.com/riya-17/pbench/pull/4.